src/modules/sensors/vehicle_acceleration/VehicleAcceleration.cpp

```diff
@@ -115,6 +113,14 @@ void VehicleAcceleration::CheckFilters()
 
 			// reset state
 			_sample_rate_incorrect_count = 0;
+			_interval_sum = 0.f;
+			_interval_count = 0.f;
+		}
+
+		// clear periodically
+		if (_interval_count > 50000) {
```

Comment:
This erases history every 50000 samples, which may cause jumps. I would rather clear every time (each 1500 samples), and update update_rate_hz with a first order low pass filter.

Reply:
Resetting each time is also fine, my only hesitation was the possible perceived delay for a filter parameter change, but I suppose that's acceptable for now.

The mean update rate over even a much smaller set of samples is quite stable other than at boot, and of course the failure scenario.